Improve domain separation guidance (#176)
openpgp-pqc/draft-openpgp-pqc: draft-ietf-openpgp-pqc.md

    //   algId           - the OpenPGP algorithm ID of the public-key encryption algorithm

    KEK = SHA3-256( mlkemKeyShare || ecdhKeyShare || ecdhCipherText || ecdhPublicKey
                    || mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep) )
    return KEK

The value `domSep` is a constant set to the UTF-8 encoding of the string "OpenPGPCompositeKDFv1", i.e.

    domSep := 4F 70 65 6E 50 47 50 43 6F 6D 70 6F 73 69 74 65 4B 44 46 76 31

`len(domSep)` is the single octet with the value equal to the octet-length of `domSep`, i.e., decimal `21`.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Author: As written in my mail, "is the one-octet hexadecimal encoding" makes no sense to me. How should the byte value differ when it is encoded in one way or the other? The difference does not exist. 

If you should mean to encode it in ASCII it's not a single octet any more.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Author: When saying: "len(domSep) := 15" it does not become clear that "15" is meant to be in hex. When changing it to "0x15", we should apply that pattern document-wide.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Author: > Also deleted this "we aligned to lamps" sentence as we will not need this anymore.

Oh yes, good spot.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Author: > Also deleted this "we aligned to lamps" sentence as we will not need this anymore.

I think this rather belongs into the PR that changes the combiner inputs.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Reviewer: I would also keep it like Falko suggested with decimal `21`.

> Note that this is in line with {{I-D.ietf-lamps-pq-composite-kem}} up to `mlkemCipherText || mlkemPublicKey || algId || domSep || len(domSep)` being the OpenPGP-specific context information.

Reviewer 2: I understand, I agree with "decimal 21".

The `domSep` information defined in {{kem-key-combiner}} provides the domain separation for the key combiner construction.
This ensures that the input keying material is used to generate a KEK for a specific purpose.

> Note that it has to be ensured that the relevant set of values of `domSep` is suffix-free when defining new values of `domSep`.

Reviewer 3: I think this is clearer:

```suggestion
When defining new values of `domSep` in the future, the set of all `domSep` values must be suffix-free.
```

But still not clear enough.  what does "suffix-free" actually mean?  they must not share a suffix?  no `domSep` value can be a suffix of any other?  Is there some registry of all `domSep` values somewhere that we could add this guidance to?

Author: A suffix-free language is one in which no word is the suffix of another. This means that in a language with this property, when parsing for a word from the rear end of a string, when finding a word, one can be sure that the found word isn't actually part of another longer word. Thus there is no ambiguity in when parsing a word of this language from the rear end. (A prefix free language gives the according property when parsing a word in from the beginning.)
